README.md

```diff
@@ -1,78 +1,137 @@
+
+```js
+printString("oops");
+// o o p s undefined
+
+printString("That's not good...");
+// T h a t '
+```
+
+It's clear that we won't be able to write out a series of string index references (`myString[0], myString[1], ...`) that can accommodate strings of varying length, so let's re-think our approach a bit.
+
+Once we `console.log()` out `myString[0]`, we won't ever need to access `myString[0]` again. The only data we need to retain are the remaining characters in `myString`. Consider this: what happens if we simply get rid of the current character at `myString[0]` and shift everything else over by one place? We could then call `myString[0]` again, but this time it would refer to the *new* first character (the character stored at `myString[1]` prior to the shift).
+
+```js
+let myString = 'pizza';
+console.log(myString[0]);
+// p
+
+myString = 'izza';
+console.log(myString[0]);
+// i
+
+myString = 'zza';
+// and so on...
 ```
 
-Now we ask if we can perform this operation by incorporating use the function printString.  We see that we can.  Print string seems to be equivalent to printing the first letter followed by printing all but the first letter of a string.  Or, in Javascript:
+Luckily for us, there's a wonderfully compact way to accomplish this wizardry with recursion. Consulting our handy three-step technique above, it appears we're onto the third and final phase: rewording the function to call itself. What if, instead of printing out multiple characters in the same pass, the function printed out `myString[0]`, shifted every remaining character one slot to the left, and then *invoked itself with the new, shortened string?!* Let's see how that would look in JavaScript:
 
-```javascript
-function printString(string){
-  let substring
-  console.log(string[0])
-  substring = string.substring(1, string.length)
-  printString(substring)
+```js
+function printString(string) {
+  let substring;
+
+  // Print out the current first character in the string.
+  console.log(string[0]);
+
+  // Store the remainder of the string in the 'substring' variable.
+  substring = string.substring(1, string.length);
+
+  // Invoke printString() from within, passing in the remainder of the previous string.
+  printString(substring);
 }
 
-printString("pizza")
+printString("pizza");
 ```
 
-Now there is only one thing left to do, which is find the base case.  Well the stopping point should be when there is only one letter left in the string.  At that point, you can just print out the string.  So this gives us:
+Uh oh, a wild infinite loop appeared! Remember, we always need to find the base case in order to stop our recursion once its mission is accomplished. In this case, our function's recursive work will be complete once there are no remaining letters to shift to the left, or, in other words, once we're down to the last character in the string. At that point, we can just print out the final character and exit!
 
-```javascript
+```js
```

Comment:
I know it works, but I think this is why it works: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L2020